test/export_to_xml.cpp

```diff
@@ -0,0 +1,284 @@
+        case V4L2_CID_AUTOGAIN:
+            parameter->SetAttribute("name","auto_gain");
+            set_type(query_ext_ctrl.type,parameter);
+            parameter->SetAttribute("default",std::to_string((uint32_t)query_ext_ctrl.default_value));
+            parameter->SetAttribute("min",std::to_string(query_ext_ctrl.minimum));
+            parameter->SetAttribute("max",std::to_string(query_ext_ctrl.maximum));
+            parameter->SetAttribute("step",std::to_string(query_ext_ctrl.step));
+            msg1->LinkEndChild( new TiXmlText( "Automatic gain" ));
+            if(query_ext_ctrl.default_value==1){
+            msg3 = new TiXmlElement( "exclusions" );
+            parameter->LinkEndChild( msg3 );
+            msg4 = new TiXmlElement( "exclude" );
+            msg4->LinkEndChild( new TiXmlText( "gain" ));
+            msg3->LinkEndChild( msg4 );}
+            break;
+        case V4L2_CID_HFLIP:
+            parameter->SetAttribute("name","horizontal_flip");
+            set_type(query_ext_ctrl.type,parameter);
+            parameter->SetAttribute("default",std::to_string((uint32_t)query_ext_ctrl.default_value));
+            parameter->SetAttribute("min",std::to_string(query_ext_ctrl.minimum));
+            parameter->SetAttribute("max",std::to_string(query_ext_ctrl.maximum));
+            parameter->SetAttribute("step",std::to_string(query_ext_ctrl.step));
+            msg1->LinkEndChild( new TiXmlText( "Horizontal Flip" ));
+            break;
+        case V4L2_CID_VFLIP:
+            parameter->SetAttribute("name","vertical_flip");
+            set_type(query_ext_ctrl.type,parameter);
+            parameter->SetAttribute("default",std::to_string((uint32_t)query_ext_ctrl.default_value));
+            parameter->SetAttribute("min",std::to_string(query_ext_ctrl.minimum));
+            parameter->SetAttribute("max",std::to_string(query_ext_ctrl.maximum));
+            parameter->SetAttribute("step",std::to_string(query_ext_ctrl.step));
+            msg1->LinkEndChild( new TiXmlText( "Vertical Flip" ));
+            break;
+        default:
+            parameter->SetAttribute("name",(char*)query_ext_ctrl.name);
+            set_type(query_ext_ctrl.type,parameter);
+            parameter->SetAttribute("default",std::to_string((uint32_t)query_ext_ctrl.default_value));
+	    parameter->SetAttribute("min",std::to_string(query_ext_ctrl.minimum));
+	    parameter->SetAttribute("max",std::to_string(query_ext_ctrl.maximum));
+	    parameter->SetAttribute("step",std::to_string(query_ext_ctrl.step));
+            if(query_ext_ctrl.type==3)
+              enumerate_menu (fd,parameter);
+            msg1->LinkEndChild( new TiXmlText( (char*)query_ext_ctrl.name ));
+
+        }
+}
+}
+    query_ext_ctrl.id |= V4L2_CTRL_FLAG_NEXT_CTRL | V4L2_CTRL_FLAG_NEXT_COMPOUND;
+}
+if (errno != EINVAL) {
+    perror("VIDIOC_QUERY_EXT_CTRL");
+    exit(EXIT_FAILURE);
+}
+       
+doc.LinkEndChild( root );
+doc.SaveFile( "camera-def-file.xml" );
```

Comment:
hardcoded output file name.

Reply:
Check result of SaveFile

Reply:
That should be a default and definitely a command line option